redshift_sqlalchemy/dialect.py

```diff
@@ -700,70 +700,189 @@ def _get_all_constraint_info(self, connection, **kw):
+                 escape=False, allow_overwrite=False, parallel=True):
+
+        if delimiter is not None and len(delimiter) != 1:
+            raise ValueError(
+                '"delimiter" parameter must be a single character'
+            )
+
+        credentials = process_aws_credentials(
+            access_key_id=access_key_id,
+            secret_access_key=secret_access_key,
+            session_token=session_token,
+        )
+
         self.select = select
         self.unload_location = unload_location
-        self.access_key = access_key
-        self.secret_key = secret_key
-        self.session_token = session_token
-        self.options = options
+        self.credentials = credentials
+        self.manifest = manifest
+        self.delimiter = delimiter
+        self.fixed_width = fixed_width
+        self.encrypted = encrypted
+        self.gzip = gzip
+        self.add_quotes = add_quotes
+        self.null = null
+        self.escape = escape
+        self.allow_overwrite = allow_overwrite
+        self.parallel = parallel
 
 
 @compiles(UnloadFromSelect)
 def visit_unload_from_select(element, compiler, **kw):
-    ''' Returns the actual sql query for the UnloadFromSelect class
-    '''
-    return """
-           UNLOAD ('%(query)s') TO '%(unload_location)s'
-           CREDENTIALS 'aws_access_key_id=%(access_key)s;aws_secret_access_key=%(secret_key)s%(session_token)s'
-           DELIMITER '%(delimiter)s'
-           %(add_quotes)s
-           %(null_as)s
-           ALLOWOVERWRITE
-           PARALLEL %(parallel)s;
-           """ % \
-           {'query': compiler.process(element.select, unload_select=True, literal_binds=True),
-            'unload_location': element.unload_location,
-            'access_key': element.access_key,
-            'secret_key': element.secret_key,
-            'session_token': ';token=%s' % element.session_token if element.session_token else '',
-            'add_quotes': 'ADDQUOTES' if bool(element.options.get('add_quotes', True)) else '',
-            'null_as': ("NULL '%s'" % element.options.get('null_as')) if element.options.get('null_as') else '',
-            'delimiter': element.options.get('delimiter', ','),
-            'parallel': element.options.get('parallel', 'ON')}
+    """Returns the actual sql query for the UnloadFromSelect class."""
+
+    template = """
+       UNLOAD (:select) TO :unload_location
+       CREDENTIALS :credentials
+       {manifest}
+       {delimiter}
+       {encrypted}
+       {fixed_width}
+       {gzip}
+       {add_quotes}
+       {null}
+       {escape}
+       {allow_overwrite}
+       {parallel}
+    """
+    el = element
+
+    qs = template.format(
+        manifest='MANIFEST' if el.manifest else '',
+        delimiter='DELIMITER AS :delimiter' if el.delimiter is not None else '',
+        encrypted='ENCRYPTED' if el.encrypted else '',
+        fixed_width=(
+            'FIXEDWIDTH AS :fixed_width' if el.fixed_width is not None else ''
```

Comment:
Why check `None` specifically here rather than relying on the boolean interpretation of `el.fixed_width`? Would we want to omit this clause in the case of an empty list?